### ISSUE TITLE
[multi stage] Measure threadCpuTime metric (similar to existing engine) in the operators, aggregate and include in response

### ISSUE THREAD

**Issue by siddharthteotia** [apache/pinot#9414]
In the current engine, we measure threadCpuTime metric at the BaseCombineOperator level. Aggregated metric is sent to servers and broker aggregates across servers and includes in query response stats. We need to do something similar (granularity etc to be discussed) in the multi-stage engine.

**Comment by walterddr** [comment]
good idea. we also had some feedback from the community on this. but the stats definition is the first to sort out.

**Comment by siddharthteotia** [comment]
I agree. Measurement is probably ok but correct propagation, aggregation and reflecting at different stages is also something that needs to be thought about
